janus_core/helpers/utils.py

```diff
@@ -337,3 +338,168 @@ def output_structs(
     if write_results:
         write_kwargs.setdefault("write_results", not invalidate_calc)
         write(images=images, **write_kwargs)
+
+
+def write_table(
+    fmt: Literal["ascii", "csv"],
+    file: Optional[TextIO] = None,
+    units: Optional[dict[str, str]] = None,
+    formats: Optional[dict[str, str]] = None,
+    **columns,
+) -> Optional[StringIO]:
+    """
+    Dump a table in a standard format.
+
+    Table columns are passed as kwargs with an optional
+    "kwarg_units" to define units for the kwarg.
+
+    Parameters
+    ----------
+    fmt : {'ascii', 'csv'}
+        Format to write table in.
+    file : TextIO
+        File to dump to. If unspecified function returns
+        io.StringIO object simulating file.
+    units : dict[str, str]
+        Units as ``{key: unit}``:
+
+        key
+            To align with those in `columns`.
+        unit
+            String defining unit.
+
+        Units which do not match any columns in `columns` are
+        ignored.
+    formats : dict[str, str]
+        Output formats as ``{key: format}``:
+
+        key
+            To align with those in `columns`.
+        format
+            Python magic format string to use.
+    **columns : dict[str, Any]
+        Dictionary of columns to columns with optional
+        "kwarg_units" to define units in column header.
+
+        See: Examples.
+
+    Returns
+    -------
+    Optional[StringIO]
+        If no file given write columns to StringIO.
+
+    Examples
+    --------
+    >>> data = write_table(fmt="ascii", single=(1, 2, 3),
+    ...                    double=(2,4,6), double_units="THz")
+    >>> print(*data, sep="", end="")
+    # single | double [THz]
+    1 2
+    2 4
+    3 6
+    >>> data = write_table(fmt="csv", a=(3., 5.), b=(11., 12.),
+    ...                    units={'a': 'eV', 'b': 'Ha'})
```

Comment:
Do we definitely want two ways to specify the units? It's a bit confusing to me, and it's not obvious that `_units` overwrites the units dict.

I don't mind too much either way, a quick comment might be sufficient